File: dictionarys_3/fantasy_game_inventory.py
import time


def display_inventory(inventory):
    print("Inventory:")
    for item, count in inventory.items():
        print(f"{item}: {count}")
    total_items = sum(inventory.values())
    print(f"Total number of items: {total_items}")


def add_inventory(inventory, loot):
    for item in loot:
        inventory[item] = inventory.get(item, 0) + 1


start_time = time.time()

inventory = {"rope": 1, "torch": 6, "gold coin": 42, "dagger": 1, "arrow": 12}

dragon_loot = ["gold coin", "dagger", "gold coin", "gold coin", "ruby"]

display_inventory(inventory)
add_inventory(inventory, dragon_loot)
display_inventory(inventory)

elapsed_time = time.time() - start_time
print(f"Time taken: {elapsed_time:.6f} seconds")

File: dictionarys_3/src/bin/fantasy_game_inventory.rs
use std::collections::HashMap;
use std::time::Instant;

fn display_inventory(inventory: &HashMap<String, i32>) {
    println!("Inventory:");
    for (item, count) in inventory {
        println!("{}: {}", item, count);
    }
    let total_items: i32 = inventory.values().sum();
    println!("Total number of items: {}", total_items);
}

fn add_inventory(inventory: &mut HashMap<String, i32>, loot: &[&str]) {
    for item in loot {
        let count = inventory.entry(item.to_string()).or_insert(0);
        *count += 1;
    }
}

fn main() {
    let start_time = Instant::now();
    let mut inventory = HashMap::from([
        ("rope".to_string(), 1),
        ("torch".to_string(), 6),
        ("gold coin".to_string(), 42),
        ("dagger".to_string(), 1),
        ("arrow".to_string(), 12),
    ]);
    const DRAGON_LOOT: [&str; 5] = ["gold coin", "dagger", "gold coin", "gold coin", "ruby"];
    display_inventory(&inventory);
    add_inventory(&mut inventory, &DRAGON_LOOT);
    display_inventory(&inventory);

    let elapsed_time = start_time.elapsed();

    println!("Time taken: {:.6} seconds", elapsed_time.as_secs_f64());
}
